core/hosting/HostWithCoreClrHost/src/SampleHost.cpp

```diff
@@ -0,0 +1,324 @@
+    // <Snippet1>
+    HMODULE coreClr = LoadLibraryExA(coreClrPath.c_str(), NULL, 0);
+    // </Snippet1>
+#elif LINUX 
+    void *coreClr = dlopen(coreClrPath.c_str(), RTLD_NOW | RTLD_LOCAL);
+#endif 
+    if (coreClr == NULL)
+    {
+        printf("ERROR: Failed to load CoreCLR from %s\n", coreClrPath.c_str());
+        return -1;
+    }
+    else
+    {
+        printf("Loaded CoreCLR from %s\n", coreClrPath.c_str());
+    }
+
+    //
+    // STEP 2: Get CoreCLR hosting functions
+    //
+#if WINDOWS
+    // <Snippet2>
+    coreclr_initialize_ptr initializeCoreClr = (coreclr_initialize_ptr)GetProcAddress(coreClr, "coreclr_initialize");
+    coreclr_create_delegate_ptr createManagedDelegate = (coreclr_create_delegate_ptr)GetProcAddress(coreClr, "coreclr_create_delegate");
+    coreclr_shutdown_ptr shutdownCoreClr = (coreclr_shutdown_ptr)GetProcAddress(coreClr, "coreclr_shutdown");
+    // </Snippet2>
+#elif LINUX    
+    coreclr_initialize_ptr initializeCoreClr = (coreclr_initialize_ptr)dlsym(coreClr, "coreclr_initialize");
+    coreclr_create_delegate_ptr createManagedDelegate = (coreclr_create_delegate_ptr)dlsym(coreClr, "coreclr_create_delegate");
+    coreclr_shutdown_ptr shutdownCoreClr = (coreclr_shutdown_ptr)dlsym(coreClr, "coreclr_shutdown");
+#endif
+
+    if (initializeCoreClr == NULL)
+    {
+        printf("coreclr_initialize not found");
+        return -1;
+    }
+
+    if (createManagedDelegate == NULL)
+    {
+        printf("coreclr_create_delegate not found");
+        return -1;
+    }
+
+    if (shutdownCoreClr == NULL)
+    {
+        printf("coreclr_shutdown not found");
+        return -1;
+    }
+
+    //
+    // STEP 3: Construct AppDomain properties used when starting the runtime
+    //
+
+    // Construct the trusted platform assemblies (TPA) list
+    // This is the list of assemblies that .NET Core can load as
+    // trusted system assemblies (similar to the .NET Framework GAC).
+    // For this host (as with most), assemblies next to CoreCLR will 
+    // be included in the TPA list
+    std::string tpaList;
+    BuildTpaList(runtimePath, ".dll", tpaList);
+
+    // <Snippet3A>
+    // Define CoreCLR properties
+    const char* propertyKeys[] = {
+        "TRUSTED_PLATFORM_ASSEMBLIES",      // Trusted assemblies (like the GAC)
+        "APP_PATHS",                        // Directories to probe for application assemblies
+        // "APP_NI_PATHS",                     // Directories to probe for application native images (not used in this sample)
+        // "NATIVE_DLL_SEARCH_DIRECTORIES",    // Directories to probe for native dlls (not used in this sample)
+    };
+
+    const char* propertyValues[] = {
+        tpaList.c_str(),
+        runtimePath
+    };
+    // </Snippet3>
+
+    //
+    // STEP 4: Start the CoreCLR runtime
+    //
+
+    // <Snippet3>
+    void* hostHandle;
+    unsigned int domainId;
+
+    // This function both starts the .NET Core runtime and creates
+    // the default AppDomain
```

Comment:
I think the AppDomain reference here is useful because it contrasts coreclrhost.h with ICLRRuntimeHost2. If someone has hosted with mscoree previously, they may wonder why they don't have to create a domain. I think it's useful to explain that the default (and only) AppDomain is created as part of `initializeCoreClr`.

Reply:
OK - I can live with the one mention here :-)